api/commands/run.ts

```diff
@@ -329,6 +353,78 @@ export class Runner {
     await this.triggerChange();
     return taskResult.status;
   }
+
+  private isActionCacheable(action: dataform.IExecutionAction): boolean {
+    if (!this.graph.runConfig.useRunCache) {
+      return false;
+    }
+
+    if (action.type === "operation") {
+      return false;
+    }
+
+    for (const dependencyTarget of action.dependencyTargets) {
+      const dependencyAction = this.graph.actions.find(
+        action => action.target === dependencyTarget
+      );
+      if (!dependencyAction) {
+        continue;
+      }
+
+      if (dependencyAction.type === "operation") {
```

Comment:
it seems weird to repeat this logic. already below you check that all dependencies of this action ran, and completed with `CACHE_SKIPPED`, thus this check must have already been run for the dependency, right?